core/headerchain.go

```diff
@@ -92,12 +95,20 @@ func NewHeaderChain(db ethdb.Database, engine consensus.Engine, pEtxsRollupFetch
 	pendingEtxsRollup, _ := lru.New(c_maxPendingEtxsRollup)
 	hc.pendingEtxsRollup = pendingEtxsRollup
 
-	pendingEtxs, _ := lru.New(c_maxPendingEtxBatches)
-	hc.pendingEtxs = pendingEtxs
+	if nodeCtx == common.PRIME_CTX {
+		pendingEtxs, _ := lru.New(c_maxPendingEtxBatchesPrime)
+		hc.pendingEtxs = pendingEtxs
```

Comment:
I'm curious why you created pendingEtxs variable?

Reply:
no need, directly assinging